src/libelfmaster.c

```diff
@@ -2297,3 +2298,351 @@ elf_close_object(elfobj_t *obj)
+						phdr->p_flags |= PF_X;
+
+					} else {	
+						Elf64_Phdr *phdr = &host->phdr64[p_iter.index-1];
+						poffset = phdr->p_offset + phdr->p_filesz;
+						phdr->p_memsz += payload_size;
+						phdr->p_filesz += payload_size;
+						phdr->p_flags |= PF_X;
+					}
+					memcpy(dest_mem, host->mem, host->size);
+					memcpy(dest_mem + poffset, target->mem, target->size);
+					*payload_offset = poffset;
+					host->mem = dest_mem;
+					host->size += payload_size;
+					injection_end = true;
+					break;
+				}
+			}
+			break;
+		}
+		case ELF_INJECT_F_PREPEND: {
+			elf_segment_iterator_init(host, &p_iter);
+			while (elf_segment_iterator_next(&p_iter, &segment) == ELF_ITER_OK) {
+				if(segment.offset != host->text_offset) {
+					if (host->e_class == ELFCLASS32) {
+						Elf32_Phdr *phdr = &host->phdr32[p_iter.index-1];
+						phdr->p_offset += payload_size;
+					} else {
+						Elf64_Phdr *phdr = &host->phdr64[p_iter.index-1];
+						phdr->p_offset += payload_size;
+					}
+				} else if(segment.offset == host->text_offset && segment.type == PT_LOAD){
+					if (host->e_class == ELFCLASS32) {
+						Elf32_Phdr *phdr = &host->phdr32[p_iter.index-1];
+						phdr->p_vaddr -= payload_size;
+						phdr->p_paddr -= payload_size;
+						phdr->p_filesz += payload_size;
+						phdr->p_memsz += payload_size;
+					} else {
+						Elf64_Phdr *phdr = &host->phdr64[p_iter.index-1];
+						phdr->p_vaddr -= payload_size;
+						phdr->p_paddr -= payload_size;
+						phdr->p_filesz += payload_size;
+						phdr->p_memsz += payload_size;
+					}
+				}
+			}
+			elf_section_iterator_init(host, &s_iter);
+			while (elf_section_iterator_next(&s_iter, &section) == ELF_ITER_OK) {
+				if (host->e_class == ELFCLASS32) {
+					Elf32_Shdr *shdr = &host->shdr32[s_iter.index-1];
+					shdr->sh_offset += payload_size;
+				} else {
+					Elf64_Shdr *shdr = &host->shdr64[s_iter.index-1];
+					shdr->sh_offset += payload_size;
+				}
+			}		
+			memcpy(dest_mem, host->mem, ehdr_size);
+			memcpy(dest_mem + ehdr_size, target->mem, target->size);
+			memcpy(dest_mem + ehdr_size + payload_size, host->mem + ehdr_size, host->size - ehdr_size);
+			
+			*payload_offset = ehdr_size;
+			host->mem = dest_mem;
+			host->size += payload_size;
+			if (host->e_class == ELFCLASS32) {
+				Elf32_Ehdr *ehdr = (Elf32_Ehdr*)host->mem;
+				ehdr->e_phoff += payload_size;
+				ehdr->e_shoff += payload_size;
+			} else {
+				Elf64_Ehdr *ehdr = (Elf64_Ehdr*)host->mem;
+				ehdr->e_phoff += payload_size;
+				ehdr->e_shoff += payload_size;
+			}
+			injection_end = true;
+			break;	
+		}
+		case ELF_INJECT_F_INTERPEND: {
+			size_t code_size;
+			size_t data_size;
+			
+			elf_segment_iterator_init(host, &p_iter);
+			while (elf_segment_iterator_next(&p_iter, &segment) == ELF_ITER_OK) {
+				if (segment.offset == host->text_offset && segment.type == PT_LOAD) {
+					if (host->e_class == ELFCLASS32) {
+						Elf32_Phdr *phdr = &host->phdr32[p_iter.index-1];
```

Comment:
Make a blank line after this line before the line of code. And follow this whenever you define or declare a variable, always make sure there is a blank line before the next line of code. There are several places but I wont point them all out. Its an easy fix. Sorry to be a pain in the ass, I try to follow it to a tee myself.